docs/graphs/graph-queries/traversal-queries/filtering-paths.md

```diff
@@ -0,0 +1,84 @@
+---
+sidebar_position: 25
+title: Filtering Paths
+---
+
+Path filtering enables advanced and efficient filtering capabilities, significantly impacting query performance. Using the `path` variable, you can apply filters based on specific iteration depths. You can filter for absolute positions in the path by specifying a positive number (thus benefiting from optimization strategies), or for relative positions to the path's end by specifying a negative number.
+
+The following examples are based on the [traversal graph](../../graph-examples/example-graphs#the-traversal-graph). If create this example graph, then you can run any of the queries listed in this topic.
+
+## Filtering Edges on the Path
+
+```sql
+FOR v, e, p IN 1..5 OUTBOUND 'circles/A' GRAPH 'traversalGraph'
+    FILTER p.edges[0].theTruth == true
+    RETURN { vertices: p.vertices[*]._key, edges: p.edges[*].label }
+```
+
+This query filters all paths where the starting edge (index 0) has the attribute `theTruth` equal to `true`. The resulting paths will be up to five items long
+
+## Filtering Vertices on the Path
+
+Similar to filtering edges, you can also filter vertices:
+
+```sql
+FOR v, e, p IN 1..5 OUTBOUND 'circles/A' GRAPH 'traversalGraph'
+    FILTER p.vertices[1]._key == "G"
+    RETURN { vertices: p.vertices[*]._key, edges: p.edges[*].label }
+```
+
+## Combining Multiple Filters
+
+You can combine filters in any way you like:
+
+```sql
+FOR v, e, p IN 1..5 OUTBOUND 'circles/A' GRAPH 'traversalGraph'
+    FILTER p.edges[0].theTruth == true
+        AND p.edges[1].theFalse == false
+    FILTER p.vertices[1]._key == "G"
+    RETURN { vertices: p.vertices[*]._key, edges: p.edges[*].label }
+```
+
+This query filters all paths where the first edge has the attribute `theTruth` equal to `true`, the first vertex is "G," and the second edge has the attribute `theFalse` equal to `false`. The resulting paths will be up to five items long.
+
+:::note
+Although the query has a defined a minimum of 1, it only returns results of depth 2. This is because, for all results in depth 1, the second edge does not exist and cannot fulfill the condition.
+:::
+
+## Filtering the Entire Path
+
+Using array comparison operators, filters can also be defined for the entire path. For example, requiring that `ALL` edges have `theTruth == true`:
+
+```sql
+FOR v, e, p IN 1..5 OUTBOUND 'circles/A' GRAPH 'traversalGraph'
+    FILTER p.edges[*].theTruth ALL == true
+    RETURN { vertices: p.vertices[*]._key, edges: p.edges[*].label }
+```
+
+Or `NONE` of the edges should have `theTruth == true`:
+
+```sql
```

Comment:
@martin-macrometa  - This query returns an empty list. Is that correct? If so, can you provide an example with NONE that does return results?

Reply:
The result would depend on the data populated in the database. Having an empty result is not a bug.

Reply:
Yeah, it just looks tacky in our documentation. I'll add a note.

Reply:
I was hoping there would be an easy fix for the query to make it have results. It's based on the Traversal Graph example.